lightningd/peer_control.c

```diff
@@ -1343,6 +1343,89 @@ static const struct json_command getinfo_command = {
 };
 AUTODATA(json_command, &getinfo_command);
 
+
+static struct command_result *json_setchannelfee(struct command *cmd,
+					 const char *buffer,
+					 const jsmntok_t *obj UNNEEDED,
+					 const jsmntok_t *params)
+{
+	struct json_stream *response;
+	const jsmntok_t *idtok;
+	unsigned int *base;
+	unsigned int *ppm;
+	struct peer *peer;
+	/* FIXME: gcc 7.3.0 thinks this might not be initialized. */
+	struct channel *channel = NULL;
+	struct channel_id cid;
+	struct command_result *res;
+
+	if (!param(cmd, buffer, params,
+		   p_req("id", param_tok, &idtok),
+		   p_opt_def("base", param_number, &base, 0),
+		   p_opt_def("ppm", param_number, &ppm, 0),
+		   NULL))
```

Comment:
Hmm, should not providing one of these leave it unchanged, or return it to the global value?

I would be tempted to side-step the debate by insisting on both values for now, and as suggested before, removing the 0 == global idea.

Reply:
I also would like to return the global value if unchanged or not set, but I need to have access to these values within the context of peer_control, which currently does not have access to global `ln->config` values.

Should I introduce these changes?

Reply:
See https://github.com/ElementsProject/lightning/pull/2394 :)

Reply:
same here, if I set all channels to current global values, they will always be applied even if global values change. is this intended/expected?

Reply:
other than that, good change on the db migrations :D